vwr/driver.py

```diff
@@ -88,12 +88,28 @@ def turn_off(self):
         self.listener.settimeout(self.timeout)
         return (response == '!')
 
+    def check_fault(self):
+        """Checks for faults with the bath
+
+        Returns:
+            True if faulty, else False
+        """
+        self._send('RF')
+        response = self._receive()
+        return (response == '1')
+
     def get(self):
         """Gets the setpoint and internal temperature."""
-        response = {'setpoint': self.get_setpoint(),
-                    'actual': self.get_internal_temperature(),
-                    'pump': self.get_pump_speed(),
-                    'on': self.get_operating_status(),
+        setpoint = self.get_setpoint()
+        actual = self.get_internal_temperature()
+        pump = self.get_pump_speed()
+        on = self.get_operating_status()
+
+        response = {'setpoint': setpoint if setpoint else False,
+                    'actual': actual if actual else False,
+                    'pump': pump if pump else False,
+                    'on': on if on else False,
+                    'fault': self.check_fault(),
                     'connected': True}
         if self.connected:
             return response
```

Comment:
Here, what I'd do it catch if any of the response structure is None. If any is None, then communication cut out during the request and we should just treat it as disconnected.

So, `if self.connected and all(v is not None for v in response)`. Something like that.